components/keys/src/lib.rs

```diff
@@ -205,11 +204,26 @@ impl EncryptedMasterKey {
 /// Creates a TOML file that contains encrypted master and returns `Keys` derived from it.
 pub fn generate_keys<P: AsRef<Path>>(path: P, passphrase: &[u8]) -> Result<Keys, failure::Error> {
     let tree = SecretTree::new(&mut thread_rng());
-    save_master_key(path, passphrase, tree.seed())?;
+    let encrypted_key = EncryptedMasterKey::encrypt(tree.seed(), passphrase)?;
+    save_master_key(path, &encrypted_key)?;
     generate_keys_from_master_password(&tree)
         .ok_or_else(|| format_err!("Error deriving keys from master key."))
 }
 
+/// Creates a TOML file from seed that contains encrypted master and returns `Keys` derived from it.
+pub fn generate_keys_from_seed(
+    passphrase: &[u8],
+    seed: &[u8],
+) -> Result<(Keys, EncryptedMasterKey), failure::Error> {
+    let tree = SecretTree::from_seed(seed)
+        .ok_or_else(|| format_err!("Error creating SecretTree from seed"))?;
+    let encrypted_key = EncryptedMasterKey::encrypt(tree.seed(), passphrase)?;
+    let keys = generate_keys_from_master_password(&tree)
+        .ok_or_else(|| format_err!("Error deriving keys from master key."))?;
+
+    Ok((keys, encrypted_key))
+}
+
 fn generate_keys_from_master_password(tree: &SecretTree) -> Option<Keys> {
```

Comment:
Off-topic: I doubt that this method should really return an `Option`; calls with `?` are seemingly infallible and could be replaced with `unwrap`s (or `Seed::new` constructor could be used).

Reply:
Really good catch 👍🏻